src/utils/useChromaticDialog.ts

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect, useRef } from "react";
+
+export type DialogEventName = "login";
+export type DialogPayload<T extends DialogEventName> = Record<string, never>;
+export type DialogHandlers = {
+  [Event in DialogEventName]?: (payload: DialogPayload<Event>) => void;
+};
+
+export const useChromaticDialog = (handlers: DialogHandlers = {}) => {
+  const dialog = useRef<Window>();
+  const dialogOrigin = useRef<string>();
+
+  // Close the dialog window when the screen gets unmounted.
+  useEffect(() => () => dialog.current?.close(), []);
+
+  useEffect(() => {
+    const handleMessage = ({ origin, data }: MessageEvent) => {
+      if (origin === dialogOrigin.current) {
+        const { message, ...rest } = data;
+        if (message in handlers) {
+          handlers[message as DialogEventName](rest);
+        }
```

Comment:
Maybe it's better if `useChromaticDialog` would take a single `messageHandler(message, rest)` that receives all messages, rather than doing the filtering in here. That would avoid having an object on the `useEffect` deps, which is really easy to accidentally recreate each render. The same is true of a function, but at least `useCallback` is more common than `useMemo`.

Reply:
Hmm, but I guess there are typing advantages of passing an object. I'll see if I can make the typing work with a function `(eventName, eventData)`.

Reply:
Figuring this out with @kasperpeulen. I'd like to get the pattern right.

Reply:
You can make the typings work with a pattern like this:

<img width="695" alt="image" src="https://github.com/chromaui/addon-visual-tests/assets/1035299/9af36131-6a28-462c-93c5-51411877e495">

https://www.typescriptlang.org/play?jsx=0#code/FAFwngDgpgBAogNygOxAZxgXhgbSk1AOQEMBbKALhgHIAjAG2OoBoYJix6B7YgEyrqN2nHr2oBdGAB8YufChAlyAgGZcuLNh258qAJnHBgKgK7IAxiACWXZDBNooAYQAWAJy6li18wBErxNwA5gAULsTIvPRQblQh8qhUiApoAJRYAHwwCFxWvOkA3sCywAC+Rua2aCAwDMRY9o6uHl4+-oFcoSF4BIpkUKzCOrzi6ZhZRbJWKjDxvUqwmEs0ddSFxbIwlchoXNEAdMEhQ6KpANwbsgD0V5t3dwB6APwbpTBQ9I4w07MJfeRYZbUNQadZ3ba7A5HE58c6XGA3e5I56vMrnIA